test/topology_custom/test_view_build_status.py

```diff
@@ -0,0 +1,162 @@
+#
+import pytest
+import time
+import asyncio
+import logging
+from test.pylib.util import unique_name, wait_for_cql_and_get_hosts, wait_for, wait_for_view
+from test.pylib.manager_client import ManagerClient
+from test.pylib.internal_types import ServerInfo
+from test.topology.util import trigger_snapshot, wait_until_topology_upgrade_finishes, enter_recovery_state, reconnect_driver, \
+        delete_raft_topology_state, delete_raft_data_and_upgrade_state, wait_until_upgrade_finishes
+from test.topology.conftest import skip_mode
+from cassandra import ConsistencyLevel
+from cassandra.query import SimpleStatement
+from cassandra.protocol import InvalidRequest
+
+
+logger = logging.getLogger(__name__)
+
+async def create_keyspace(cql):
+    ks_name = 'ks'
+    await cql.run_async(f"CREATE KEYSPACE {ks_name} WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}}")
+    return ks_name
+
+async def create_table(cql):
+    await cql.run_async(f"CREATE TABLE ks.t (p int, c int, PRIMARY KEY (p, c))")
+
+async def create_mv(cql, view_name):
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.{view_name} AS SELECT * FROM ks.t WHERE c IS NOT NULL and p IS NOT NULL PRIMARY KEY (c, p)")
+
+async def get_view_builder_version(cql, **kwargs):
+    result = await cql.run_async("SELECT value FROM system.scylla_local WHERE key='view_builder_version'", **kwargs)
+    if len(result) == 0:
+        return 1
+    else:
+        return int(result[0].value)
+
+async def view_builder_is_v2(cql, **kwargs):
+    v = await get_view_builder_version(cql, **kwargs)
+    return v == 2 or None
+
+async def view_is_built_v2(cql, ks_name, view_name, node_count, **kwargs):
+    done = await cql.run_async(f"SELECT COUNT(*) FROM system.view_build_status_v2 WHERE keyspace_name='{ks_name}' \
+                                 AND view_name = '{view_name}' AND status = 'SUCCESS' ALLOW FILTERING", **kwargs)
+    return done[0][0] == node_count or None
+
+async def wait_for_view_v2(cql, ks_name, view_name, node_count, **kwargs):
+    await wait_for(lambda: view_is_built_v2(cql, ks_name, view_name, node_count, **kwargs), time.time() + 60)
+
+# Verify a new cluster uses the view_build_status_v2 table.
+# Create a materialized view and check that the view's build status
+# is stored in view_build_status_v2 and all nodes see all the other
+# node's statuses.
+@pytest.mark.asyncio
+async def test_view_build_status_v2_table(manager: ManagerClient):
+    node_count = 3
+    servers = await manager.servers_add(node_count)
+    cql, hosts = await manager.get_ready_cql(servers)
+
+    # The cluster should init with view_builder v2
+    for h in hosts:
+        v = await get_view_builder_version(cql, host=h)
+        assert v == 2
+
+    await create_keyspace(cql)
+    await create_table(cql)
+    await create_mv(cql, "vt")
+
+    await asyncio.gather(*(wait_for(lambda: view_is_built_v2(cql, 'ks', 'vt', node_count, host=h), time.time() + 60) for h in hosts))
```

Comment:
Nit: why not use `wait_for_view_v2` directly?

Reply:
done